src/main/scala/org/broadinstitute/dsde/firecloud/core/ProfileClient.scala

```diff
@@ -0,0 +1,130 @@
+package org.broadinstitute.dsde.firecloud.core
+
+import akka.actor.{Actor, Props}
+import akka.event.Logging
+import akka.pattern.pipe
+
+import org.broadinstitute.dsde.firecloud.core.ProfileClient.UpdateProfile
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.broadinstitute.dsde.firecloud.model._
+import org.broadinstitute.dsde.firecloud.service.{UserService, FireCloudRequestBuilding}
+import org.broadinstitute.dsde.firecloud.service.PerRequest.{PerRequestMessage, RequestComplete}
+
+import spray.client.pipelining._
+import spray.http.HttpResponse
+import spray.http.StatusCodes._
+import spray.httpx.SprayJsonSupport._
+import spray.routing.RequestContext
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+object ProfileClient {
+  case class UpdateProfile(userInfo: UserInfo, profile: Profile)
+  def props(requestContext: RequestContext): Props = Props(new ProfileClientActor(requestContext))
+}
+
+class ProfileClientActor(requestContext: RequestContext) extends Actor with FireCloudRequestBuilding {
+
+  implicit val system = context.system
+  import system.dispatcher
+  val log = Logging(system, getClass)
+
+  override def receive: Receive = {
+
+    case UpdateProfile(userInfo: UserInfo, profile: Profile) =>
+      val parent = context.parent
+      val pipeline = authHeaders(requestContext) ~> sendReceive
```

Comment:
I refactored the other actor classes, but not the OAuth one yet. Have a look at those. In general, where we are using 
`future onComplete ...`, we should instead be constructing a Future[PerRequest] somewhere and then piping that to the context.parent. The problem we had was that the context was not fixed and in between actor calls, the context as seen by one future block could be a different context that initially created it. Not exactly a thread-safety issue, but a similar effect. 

Looking at pipeline, I think it's OK if we fix it as a val at the beginning of the case class and then pass it around. That way the requestContext used in the various futures is guaranteed to be the same one that initiated the actor call.